neo/SmartContract/ApplicationEngine.cs

```diff
@@ -100,17 +112,111 @@ private bool CheckItemSize()
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        private bool CheckBigInteger(BigInteger value)
+        {
+            return value == null ? false :
+                value.ToByteArray().Length <= MaxSizeForBigInteger;
+        }
+
+        /// <summary>
+        /// Check if the BigInteger is allowed for numeric operations
+        /// </summary> 
+        private bool CheckBigIntegers(OpCode nextInstruction)
+        {
+            switch (nextInstruction)
+            {
+                case OpCode.INC:
+                    {
+                        BigInteger x = EvaluationStack.Peek().GetBigInteger();
+
+                        if (!CheckBigInteger(x) || !CheckBigInteger(x + 1))
+                            return false;
+
+                        break;
+                    }
+                case OpCode.DEC:
+                    {
+                        BigInteger x = EvaluationStack.Peek().GetBigInteger();
+
+                        if (!CheckBigInteger(x) || !CheckBigInteger(x - 1))
+                            return false;
+
+                        break;
+                    }
+                case OpCode.ADD:
+                    {
+                        BigInteger x2 = EvaluationStack.Peek().GetBigInteger();
+                        BigInteger x1 = EvaluationStack.Peek(1).GetBigInteger();
+
+                        if (!CheckBigInteger(x2) || !CheckBigInteger(x1) || !CheckBigInteger(x1 + x2))
+                            return false;
+
+                        break;
+                    }
+                case OpCode.SUB:
+                    {
+                        BigInteger x2 = EvaluationStack.Peek().GetBigInteger();
+                        BigInteger x1 = EvaluationStack.Peek(1).GetBigInteger();
+
+                        if (!CheckBigInteger(x2) || !CheckBigInteger(x1) || !CheckBigInteger(x1 - x2))
+                            return false;
+
+                        break;
+                    }
+                case OpCode.MUL:
+                    {
+                        BigInteger x2 = EvaluationStack.Peek().GetBigInteger();
+                        BigInteger x1 = EvaluationStack.Peek(1).GetBigInteger();
+
+                        int lx1 = x1 == null ? 0 : x1.ToByteArray().Length;
+
+                        if (lx1 > MaxSizeForBigInteger)
+                            return false;
+
+                        int lx2 = x2 == null ? 0 : x2.ToByteArray().Length;
+
+                        if ((lx1 + lx2) > MaxSizeForBigInteger)
+                            return false;
+
+                        break;
+                    }
+                case OpCode.DIV:
+                    {
+                        BigInteger x2 = EvaluationStack.Peek().GetBigInteger();
+                        BigInteger x1 = EvaluationStack.Peek(1).GetBigInteger();
+
+                        if (!CheckBigInteger(x2) || !CheckBigInteger(x1) || !CheckBigInteger(x1 / x2))
```

Comment:
If `CheckBigInteger(x2) == true` and `CheckBigInteger(x1) == true`, then it is impossible that `CheckBigInteger(x1 / x2)) == false`.

Reply:
You are right!